flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobExceptionsInfo.java

```diff
@@ -190,21 +193,21 @@ public boolean equals(Object o) {
             return timestamp == that.timestamp
                     && Objects.equals(exception, that.exception)
                     && Objects.equals(task, that.task)
-                    && Objects.equals(location, that.location)
+                    && Objects.equals(endpoint, that.endpoint)
```

Comment:
if an old object with location = aaaa:1234 and endpoint unset . If we compare this with a new object with endpoint = aaaa:1234 but location not set. This test would say they are unequal, but the objects have the same hash (as the toString of location and endpoint are the same I think).

Reply:
Both fields will be assigned within Flink in all cases, so maybe this is not a problem to be concerned about.